azure-pipelines.yml

```diff
@@ -32,21 +32,46 @@ steps:
     pmdRunAnalysis: true
     findBugsRunAnalysis: true
 
-# - task: Bash@3
-#   displayName: 'test example project with gradle'
-#   inputs:
-#     targetType: 'inline'
-#     script: 'rm -rf example-projects/example-project/target &&
-#       cd example-projects/example-project &&
-#       gradle analyze &&
-#       java -jar ../../core/frontend-stubs-testing/target/antenna-test-project-asserter.jar ExampleTestProject build'
-#     failOnStderr: true
+- task: Bash@3
+  displayName: 'test example project with maven'
+  inputs:
+    targetType: 'inline'
+    script: 'tmpdir=$(mktemp -d) &&
+      cp -r example-projects/example-project/ $tmpdir/example-project/ &&
+      rm -rf $tmpdir/example-project/target &&
+      pushd $tmpdir/example-project/ &&
+      mvn package &&
+      popd &&
+      java -jar core/frontend-stubs-testing/target/antenna-test-project-asserter.jar ExampleTestProject $tmpdir/example-project/target'
+
+- task: Bash@3
+  displayName: 'test example project with cli'
+  inputs:
+    targetType: 'inline'
+    script: 'tmpdir=$(mktemp -d) &&
+      cp -r example-projects/example-project/ $tmpdir/example-project/ &&
+      rm -rf $tmpdir/example-project/target &&
+      .travis/runCLI.sh $tmpdir/example-project &&
+      java -jar core/frontend-stubs-testing/target/antenna-test-project-asserter.jar ExampleTestProject $tmpdir/example-project/target'
 
-# - task: Bash@3
-#   displayName: 'test example project with cli'
-#   inputs:
-#     targetType: 'inline'
-#     script: 'rm -rf example-projects/example-project/target &&
-#       .travis/runCLI.sh example-projects/example-project &&
-#       java -jar core/frontend-stubs-testing/target/antenna-test-project-asserter.jar ExampleTestProject target'
-#     failOnStderr: true
+- task: Bash@3
+  displayName: 'test example project with gradle'
+  inputs:
+    targetType: 'inline'
+    script: 'tmpdir=$(mktemp -d) &&
+      cp -r example-projects/example-project/ $tmpdir/example-project/ &&
+      rm -rf $tmpdir/example-project/target &&
+      rm -rf $tmpdir/example-project/build &&
+      cp assembly/gradle-plugin/gradlew $tmpdir/example-project/ &&
+      cp -r assembly/gradle-plugin/gradle $tmpdir/example-project/ &&
+      pushd $tmpdir/example-project &&
+      ./gradlew analyze &&
```

Comment:
Nit, maybe `./gradlew -p $tmpdir/example-project analyze` instead of the `pushd` / `popd` sequence?

Reply:
The following does not work:
```diff
diff --git a/.ci-scripts/test-ExampleTestProject-with-gradle.sh b/.ci-scripts/test-ExampleTestProject-with-gradle.sh
index 6fb0c6cb..2f79d611 100755
--- a/.ci-scripts/test-ExampleTestProject-with-gradle.sh
+++ b/.ci-scripts/test-ExampleTestProject-with-gradle.sh
@@ -15,11 +15,9 @@ cd "$(dirname "$0")/.."
 tmpdir=$(mktemp -d)
 trap 'rm -rf $tmpdir' EXIT
 cp -r example-projects/example-project/ $tmpdir/example-project/
-rm -rf $tmpdir/example-project/target
-rm -rf $tmpdir/example-project/build
 cp assembly/gradle-plugin/gradlew $tmpdir/example-project/
 cp -r assembly/gradle-plugin/gradle $tmpdir/example-project/
-pushd $tmpdir/example-project
-./gradlew analyze
-popd
+$tmpdir/example-project/gradlew -p $tmpdir/example-project cleanAnalyze analyze
 java -jar core/frontend-stubs-testing/target/antenna-test-project-asserter.jar ExampleTestProject $tmpdir/example-project/build
```

Reply:
I guess that's because the example project does not have its own wrapper committed. You should use:

    ./gradlew -p $tmpdir/example-project cleanAnalyze analyze

Reply:
I copy the wrapper some lines above (we also have no wrapper in the project root)
```
 cp assembly/gradle-plugin/gradlew $tmpdir/example-project/
 cp -r assembly/gradle-plugin/gradle $tmpdir/example-project/
```

Reply:
So what's the issue then (please copy & paste the error message)?

Reply:
```
+ /tmp/tmp.b2hWfv0wjb/example-project/gradlew -p /tmp/tmp.b2hWfv0wjb/example-project cleanAnalyze analyze
:cleanAnalyze UP-TO-DATE
:analyze[Fatal Error] :331:29: Zeichenfolge "--" ist in Kommentaren nicht zulässig.
Error in second stage while rendering template. This also means that first stage result is invalid.
org.xml.sax.SAXParseException; lineNumber: 331; columnNumber: 29; Zeichenfolge "--" ist in Kommentaren nicht zulässig.
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:257)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:339)
	at org.eclipse.sw360.antenna.util.XmlSettingsReader.<init>(XmlSettingsReader.java:53)
	at org.eclipse.sw360.antenna.util.TemplateRenderer.getPropertiesMapFromRenderedResult(TemplateRenderer.java:111)
	at org.eclipse.sw360.antenna.util.TemplateRenderer.renderTemplate(TemplateRenderer.java:194)
	at org.eclipse.sw360.antenna.util.TemplateRenderer.renderTemplateFile(TemplateRenderer.java:84)
	at org.eclipse.sw360.antenna.util.TemplateRenderer.renderTemplateFile(TemplateRenderer.java:59)
	at org.eclipse.sw360.antenna.frontend.stub.cli.AntennaCLISettingsReader.getSettingsReader(AntennaCLISettingsReader.java:143)
	at org.eclipse.sw360.antenna.frontend.stub.cli.AntennaCLISettingsReader.readSettingsToToolConfiguration(AntennaCLISettingsReader.java:102)
	at org.eclipse.sw360.antenna.frontend.stub.cli.AbstractAntennaCLIFrontend.init(AbstractAntennaCLIFrontend.java:75)
	at org.eclipse.sw360.antenna.frontend.stub.cli.AbstractAntennaCLIFrontend.execute(AbstractAntennaCLIFrontend.java:85)
	at org.eclipse.sw360.antenna.frontend.stub.gradle.AntennaImpl.execute(AntennaImpl.java:53)
	at org.eclipse.sw360.antenna.frontend.stub.gradle.AnalyzeTask.analyze(AnalyzeTask.java:42)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:597)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
 FAILED
```

Reply:
If I cange the PWD before the `gradlew` call to `/tmp` I get the following error:
```
+ /tmp/tmp.ZcLBJlQEvq/example-project/gradlew -p /tmp/tmp.ZcLBJlQEvq/example-project cleanAnalyze analyze
:cleanAnalyze UP-TO-DATE
:analyze FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':analyze'.
> Cannot find /tmp/pom.xml

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
2 actionable tasks: 1 executed, 1 up-to-date
```

Obviously we depend on the PWD to find the file `./pom.xml` in the case of gradle execution.

Reply:
Interesting, also good to know, and thanks or testing.